tests/foreman/longrun/test_oscap.py

```diff
@@ -615,81 +721,79 @@ def test_positive_oscap_run_via_local_files(
-            .search(query={'search': 'name="Ansible Roles - Ansible Default"'})[0]
-            .id
-        )
-        job = module_target_sat.api.JobInvocation().run(
-            synchronous=False,
-            data={
-                'job_template_id': template_id,
-                'targeting_type': 'static_query',
-                'search_query': f'name = {vm.hostname}',
-            },
-        )
-        module_target_sat.wait_for_tasks(
-            f'resource_type = JobInvocation and resource_id = {job["id"]}',
-            poll_timeout=1000,
-        )
-        assert module_target_sat.api.JobInvocation(id=job['id']).read().succeeded == 1
-        assert vm.run('cat /etc/foreman_scap_client/config.yaml | grep profile').status == 0
-        # Runs the actual oscap scan on the vm/clients
-        # TODO: instead of running it on the client itself we should invoke a job from satellite
-        result = vm.execute_foreman_scap_client()
-        assert f"WARNING: Using local file '/root/{file_name}'" in result
+    result = contenthost.register(
+        module_org,
+        None,
+        ak_name[distro],
+        module_target_sat,
+        ignore_subman_errors=True,
+        force=True,
+        insecure=True,
+    )
+    assert result.status == 0, f'Failed to register host: {result.stderr}'
+    rhel_repo = rhel_repos[distro]
+    profile = profiles[distro]
+    contenthost.create_custom_repos(**rhel_repo)
+
+    proxy_id = module_target_sat.nailgun_smart_proxy.id
+    target_host = contenthost.nailgun_host
+    module_target_sat.api.AnsibleRoles().sync(
+        data={'proxy_id': proxy_id, 'role_names': [SELECTED_ROLE]}
+    )
+    role_id = (
+        module_target_sat.api.AnsibleRoles().search(query={'search': f'name={SELECTED_ROLE}'})[0].id
+    )
+    module_target_sat.api.Host(id=target_host.id).add_ansible_role(
+        data={'ansible_role_id': role_id}
+    )
+    host_roles = target_host.list_ansible_roles()
+    assert host_roles[0]['name'] == SELECTED_ROLE
+    module_target_sat.cli_factory.make_scap_policy(
+        {
+            'scap-content-id': scap_id,
+            'hostgroups': hgrp_name,
+            'deploy-by': 'ansible',
+            'name': policy_name,
+            'period': OSCAP_PERIOD['weekly'].lower(),
+            'scap-content-profile-id': scap_profile_id,
+            'weekday': OSCAP_WEEKDAY['friday'].lower(),
+            'organizations': module_org.name,
+        }
+    )
+    # The file here needs to be present on the client in order
+    # to perform the scan from the local-files.
+    contenthost.execute(f'curl -o {file_name} {download_url}')
+    module_target_sat.cli.Host.update(
+        {
+            'name': contenthost.hostname,
+            'lifecycle-environment': lifecycle_env.name,
+            'content-view': cv_name[distro],
+            'hostgroup': hgrp_name,
+            'openscap-proxy-id': default_proxy,
+            'organization': module_org.name,
+        }
+    )
```

Comment:
Similar note as above: I'd try to create the resources once and avoid updating it after the fact.